tests/test_increased_belief_case.py

```diff
@@ -0,0 +1,133 @@
+import numpy as np
+import pytest
+
+import iterative_ensemble_smoother as ies
+
+# The following tests follow the
+# posterior properties described in
+# https://ert.readthedocs.io/en/latest/theory/ensemble_based_methods.html
+a_true = 1.0
+b_true = 5.0
+number_of_parameters = 2
+number_of_observations = 45
+
+
+class LinearModel:
+    def __init__(self, a, b):
+        self.a = a
+        self.b = b
+        self.size = 2
+
+    @classmethod
+    def random(cls):
+        a_std = 2.0
+        b_std = 2.0
+        # Priors with bias
+        a_bias = 0.5 * a_std
+        b_bias = -0.5 * b_std
+
+        return cls(
+            np.random.normal(a_true + a_bias, a_std),
+            np.random.normal(b_true + b_bias, b_std),
+        )
+
+    def eval(self, x):
+        return self.a * x + self.b
+
+
+def distance(point1, point2):
+    """euclidean distance
+    :param point1: n-dimensional point (array of floats).
+    :param point2: n-dimensional point (array of floats).
+    """
+    point1 = np.asarray(point1)
+    point2 = np.asarray(point2)
+    return np.sqrt(np.sum((point1 - point2) ** 2))
+
+
+@pytest.mark.parametrize("number_of_realizations", [10, 100, 200])
+def test_that_update_for_a_linear_model(number_of_realizations):
+    true_model = LinearModel(a_true, b_true)
+
+    ensemble = [LinearModel.random() for _ in range(number_of_realizations)]
+
+    A = np.array(
+        [
+            [realization.a for realization in ensemble],
+            [realization.b for realization in ensemble],
+        ]
+    )
+    average_prior = np.average(A, axis=1)
+
+    # We use time as the x-axis and observations are at
+    # t=0,1,2...number_of_observations
+    times = np.arange(number_of_observations)
+
+    S = np.array([[realization.eval(t) for realization in ensemble] for t in times])
+
+    # When observations != true model, then ml estimates != true parameters.
+    # This gives both a more advanced and realistic test. Standard normal
+    # N(0,1) noise is added to obtain this. The randomness ensures we are not
+    # gaming the test. But the difference could in principle be any non-zero
+    # scalar.
+    observations = np.array(
+        [true_model.eval(t) + np.random.normal(0.0, 1.0) for t in times]
+    )
+
+    # Leading to fixed Maximum likelihood estimate.
+    # It will equal true values when observations are sampled without noise.
+    # It will also stay the same over beliefs.
+    mean_observations = np.average(observations)
```

Comment:
Consider using `np.mean`. When I read `np.average` I expect a weighted average because that's what it's usually used for.